### PR TITLE
1052 - Implement new task editor 3

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #1052

Screenshot 2023-09-27 alle 11 33 27
Screenshot 2023-09-27 alle 11 33 07

### CODE REVIEW

File: web/src/app/pages/create-survey/task-details/task-details.component.html [hunk 1]
```diff
@@ -41,7 +42,7 @@ <h2>
 
   <div id="add-task-buttons-list">
     <add-task-button
-      *ngFor="let taskGroup of [TaskGroup.QUESTION, TaskGroup.PHOTO, TaskGroup.DROP_PIN, TaskGroup.DRAW_AREA, TaskGroup.SUGGEST_LOI]"
+      *ngFor="let taskGroup of [TaskGroup.QUESTION, TaskGroup.PHOTO, TaskGroup.DROP_PIN, TaskGroup.DRAW_AREA, TaskGroup.CAPTURE_LOCATION]"
```
Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-details.component.html:45]: Can we move this list into the component to keep less logic here in the view template?

File: web/src/app/pages/create-survey/task-details/task-details.component.ts [hunk 2]
```diff
@@ -26,23 +26,27 @@ export enum TaskGroup {
...
 
 export const taskTypeToGroup = new Map([
   [TaskType.TEXT, TaskGroup.QUESTION],
   [TaskType.DATE, TaskGroup.QUESTION],
+  [TaskType.MULTIPLE_CHOICE, TaskGroup.QUESTION],
   [TaskType.PHOTO, TaskGroup.PHOTO],
-  [TaskType.DRAW_AREA, TaskGroup.DRAW_AREA],
   [TaskType.DROP_PIN, TaskGroup.DROP_PIN],
+  [TaskType.DRAW_AREA, TaskGroup.DRAW_AREA],
+  [TaskType.CAPTURE_LOCATION, TaskGroup.CAPTURE_LOCATION],
```
Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-details.component.ts:49]: Should these arrays also be immutable `List`s?

File: web/src/app/pages/create-survey/task-details/task-details.component.ts [hunk 3]
```diff
@@ -110,6 +111,29 @@ export class TaskDetailsComponent {
       });
   }
 
+  onTaskDuplicate(index: number) {
```
Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-details.component.ts:114]: Nit: Word order: `onDuplicateTask()`

Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-details.component.ts:114]: Nit: Word order: `onDuplicateTask()`

File: web/src/app/pages/create-survey/task-details/task-details.component.ts [hunk 4]
```diff
@@ -110,6 +111,29 @@ export class TaskDetailsComponent {
...
+    this.dialogService
+      .openConfirmationDialog(
+        'Warning',
+        'Are you sure you wish to duplicate this task? Any associated data'
```
Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-details.component.ts:118]: If this just copies the task into a new one and doesn't delete any data, I don't think we need a warning here.

File: web/src/app/pages/create-survey/task-details/task-input/option-editor/_option-editor.component-theme.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
+@use 'sass:map';
```
Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-input/option-editor/_option-editor.component-theme.scss:1]: Please add copyright header with "The Ground Authors." as copyright owner.

File: web/src/app/pages/create-survey/task-details/task-input/option-editor/option-editor.component.html [hunk 6]
```diff
@@ -0,0 +1,54 @@
+<!--
+  Copyright 2020 Google LLC
```
Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-input/option-editor/option-editor.component.html:2]: This should read:

`  Copyright 2023 The Ground Authors.`

Same comment for all new files.

File: web/src/app/pages/create-survey/task-details/task-input/option-editor/option-editor.component.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  Validators,
+  AbstractControl,
+} from '@angular/forms';
+
+@Component({
+  selector: 'ground-option-editor',
+  templateUrl: './option-editor.component.html',
+  styleUrls: ['./option-editor.component.scss'],
+})
+export class OptionEditorComponent implements OnInit, OnChanges {
```
Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-input/option-editor/option-editor.component.ts:39]: Please name `EditOptionComponent` for consistency with `EditSurvey` and `EditJob`. The only other component with this naming convention is `InlineEditorComponent`, but that will be removed in the latest design.

**Review by gino-m — CHANGES_REQUESTED** [review]
Nice work, a few small edits/suggestions.

File: web/src/app/pages/create-survey/task-details/task-details.component.ts (hunk 2 above)
Comment (rfontanarosa) [reply web/src/app/pages/create-survey/task-details/task-details.component.ts:49]: no, they can't. Map constructor wants an Array

File: web/src/app/pages/create-survey/task-details/task-input/option-editor/_option-editor.component-theme.scss (hunk 5 above)
Comment (rfontanarosa) [reply web/src/app/pages/create-survey/task-details/task-input/option-editor/_option-editor.component-theme.scss]: all the other _*.component-theme files haven't the copyright. can we do it in a separate pr?

**Review by gino-m — APPROVED** [review]

**Pull request merged**